austrakka/components/sequence/funcs.py

```diff
@@ -101,18 +112,33 @@ def _fastq_file_2_is_present(row: pd.Series) -> bool:
     return str(row[FASTQ_CSV_PATH_2]) not in STR_NA_VALUES
 
 
-def _get_file(filepath: str) -> tuple[str, tuple[str, BinaryIO]]:
+def _get_file(filepath: str) -> SeqFile:
     # pylint: disable=consider-using-with
     file = open(filepath, 'rb')
-    return 'files[]', (os.path.basename(file.name), file)
+    filename = os.path.basename(file.name)
+    return SeqFile(
+        multipart=('files[]', (filename, file)),
+        sha256=hashlib.sha256(file.read()).hexdigest(),
+        filename=filename,
+    )
 
 
-def _post_fastq(sample_files, custom_headers):
-    api_post_multipart(
+def _post_fastq(sample_files: list[SeqFile], custom_headers):
+    files = [file.multipart for file in sample_files]
+    resp = api_post_multipart(
         path="/".join([SEQUENCE_PATH, FASTQ_PATH]),
-        files=sample_files,
+        files=files,
         custom_headers=custom_headers,
     )
+    for seq in resp['data']:
+        if not any(
+                f.filename == seq['originalFileName']
+                and f.sha256.casefold() == seq['serverSha256'].casefold()
+                for f in sample_files
+        ):
+            raise IncorrectHashException(
```

Comment:
Since we're processing more than one file (I think it's max 2 reads), we should process all the non matches in the batch and then raise the error outside the loop. Similar to what we do for metadata uploads.